libs/src/services/solanaWeb3Manager/utils.js

```diff
@@ -148,7 +148,7 @@ class SolanaUtils {
    */
   static ethAddressToArray (ethAddress) {
     const strippedEthAddress = ethAddress.replace('0x', '')
-    return Uint8Array.of(...new BN(strippedEthAddress, 'hex').toArray('be'))
+    return Uint8Array.of(...new BN(strippedEthAddress, 'hex').toArray('be', 20))
```

Comment:
why was this necessary? 

does strippedEthaddres.length make more sense (?)

Reply:
I'll add a comment - all eth arrays need to be length 20, but if your hex representation starts with some number of zeros, the resulting array will be < 20 - so need to pad with zeros